src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -661,18 +662,20 @@ private static void ReportBinaryOperatorError(ExpressionSyntax node, DiagnosticB
         {
             bool leftDefault = left.IsLiteralDefault();
             bool rightDefault = right.IsLiteralDefault();
-            if ((operatorToken.Kind() == SyntaxKind.EqualsEqualsToken || operatorToken.Kind() == SyntaxKind.ExclamationEqualsToken))
+            if (operatorToken.Kind() != SyntaxKind.EqualsEqualsToken && operatorToken.Kind() != SyntaxKind.ExclamationEqualsToken)
             {
-                if (leftDefault && rightDefault)
+                if (leftDefault || rightDefault)
                 {
-                    Error(diagnostics, ErrorCode.ERR_AmbigBinaryOpsOnDefault, node, operatorToken.Text);
+                    // other than == and !=, binary operators are disallowed on `default` literal
+                    Error(diagnostics, ErrorCode.ERR_BadOpOnTypelessExpression, node, operatorToken.Text, "default");
                     return;
                 }
             }
-            else if (leftDefault || rightDefault)
+
+            if ((leftDefault || left.IsTypelessNew()) &&
+                (rightDefault || right.IsTypelessNew()))
             {
```

Comment:
Please add a PROTOTYPE comment that this should be rewritten in terms of the operand having a `null` type.